### ISSUE TITLE
Cannot append text to the end of editor content in `instanceReady`

### ISSUE THREAD

**Issue by earshinov** [ckeditor/ckeditor-dev#1971]
## Type of report

Bug

## Reproduction steps

Please see JSFiddle: https://jsfiddle.net/yo40byby/

The task is to append some text to the end of editor content in `instanceReady` (*).

(*) In case it sounds too contrived, in the real world we have a contenteditable area where CKEDITOR is initialized lazily in order to be economical with browser resources. When a user performs a certain action, we initialize the editor if it is not already initialized and append some text appropriate for the performed action. 

The code is basically

```
var sel = ckEditor.getSelection(/* forceRealSelection = */ true);
var range = ckEditor.createRange();
range.moveToPosition(ckEditor.editable(), CKEDITOR.POSITION_BEFORE_END);
sel.selectRanges([range]);
ckEditor.insertText(text);
```

but it results in text inserted at the beginning, not at the end of editor's content.

![observed result](https://user-images.githubusercontent.com/52021/39958667-c062c668-560e-11e8-8b99-0d9591c99d4d.png)

If there is a proper way of adjusting the application code to circumvent this problem, I would very much like to hear it,

## Other details

* Browser: Google Chrome 66.0.3359.139 (Official Build) (64-bit)
* OS: Windows 7 64-bit
* CKEditor version: Reproducible in 4.9.2 (see the JSFiddle) and current master.
* Installed CKEditor plugins: "Standard" distribution.

**Comment by earshinov** [comment]
Forgot to mention, at the moment we are using CKEDITOR with `ENTER_BR` mode, but the issue reproduces both with `ENTER_BR` and `ENTER_P`.

**Comment by Comandeer** [comment]
I simplified case: https://jsfiddle.net/hja2qojd/ – what's interesting now it also fails in Firefox. However the fail occurs only during the first text insertion.

It seems that workaround is very simple: focus editor before selecting range → https://jsfiddle.net/hja2qojd/1/ https://jsfiddle.net/yo40byby/3/

Despite workaround being so simple, we'll further investigate this bug.

**Comment by earshinov** [comment]
My guess is that `fixInitialSelection` is to blame. If I redefine it as a no-op, everything works as expected without calling `editor.focus()` first: https://jsfiddle.net/hja2qojd/2/.

**Comment by Comandeer** [comment]
You're right. It seems that https://github.com/ckeditor/ckeditor-dev/blob/8c7aa8e3a50766493f85f80c67bf905eba45ad9d/core/editable.js#L584-L587 is to blame. Probably function used to determine if initial selection fix is needed needs fixing itself.
